### PR TITLE
Properly support `enter` & `check` in `mkNpmProject`

### LINKED ISSUE #229: Figure out an easy way to run `tsc` as a `Check` and within `garn enter`
#227 demonstrates a few issues that we would like to work. Maybe there's some changes to how people can achieve them, but currently it's not clear these things are doable at all. This includes being able to do:

- `garn enter default` and then `tsc` (or `npm install ; tsc`),
- Putting .addCheck("tsc")`tsc` into the `garn.ts` file and then having it run with `garn check default`.

### PR DESCRIPTION
**Pull request by soenkehahn**
Fixes #229.

### CODE REVIEW

File: examples/monorepo/garn.ts [hunk 1]
```diff
@@ -18,11 +18,9 @@ export const npmFrontend = garn.javascript.mkNpmProject({
...
   src: "frontend-npm",
   nodeVersion: "18",
-  testCommand: "npm run test -- --watchAll=false",
 });
 
 export const startAll = garn.processCompose({
   backend: backend.defaultExecutable!,
-  'yarn frontend': yarnFrontend.defaultExecutable!,
-  'npm frontend': npmFrontend.defaultExecutable!,
+  frontend: npmFrontend.devShell.shell`cd frontend-npm && npm install && npm start`,
```
Comment (soenkehahn) [inline examples/monorepo/garn.ts:25]: We don't run the yarn frontend here anymore?

**Review by soenkehahn — COMMENTED** [review]
Apart from my one comment, this LGTM for now. What do you think, @alexdavid?

File: examples/monorepo/garn.ts (hunk 1 above)
Comment (alexdavid) [reply examples/monorepo/garn.ts:25]: I figured maybe it was best to omit it for now until our yarn API becomes more stable. Happy to add it back if you feel differently though!

**Review by alexdavid — APPROVED** [review]
This looks great!

File: examples/monorepo/garn.ts (hunk 1 above)
Comment (soenkehahn) [reply examples/monorepo/garn.ts:25]: I think you can do ```npmFrontend.shell`...```.

Comment (soenkehahn) [reply examples/monorepo/garn.ts:25]: And maybe replace the yarn project with the haskell one?

**Pull request merged**